src/wallet/api/wallet2_api.h

```diff
@@ -1144,6 +1149,545 @@ struct Wallet
+    * param: key_image - key image of enote
+    * note: sets status error on fail
+    */
+    virtual void freeze(std::size_t idx) = 0;
+    virtual void freeze(const std::string &key_image) = 0;
+    // QUESTION : Should we add the option to freeze by tx_id? Meaning: freeze all enotes that we received by a single transaction.
+//    virtual void freeze(const std::string &tx_id) = 0;
+    /**
+    * brief: thaw - thaw enote that is frozen, so it appears in balance and can be spent in a transaction
+    * param: idx - index of enote in `m_transfers`
+    * param: key_image - key image of enote
+    * note: sets status error on fail
+    */
+    virtual void thaw(std::size_t idx) = 0;
+    virtual void thaw(const std::string &key_image) = 0;
+    // QUESTION : Should we add the option to thaw by tx_id? Meaning: thaw all enotes that we received by a single transaction.
+//    virtual void thaw(const std::string &tx_id) = 0;
+    /**
+    * brief: isFrozen - check if enote is frozen
+    * param: idx - index of enote in `m_transfers`
+    * param: key_image - key image of enote
+    * QUESTION : Which approach should we use? We probably don't need both. Or are there other suggestions? (See WalletImpl::isFrozen for implementation)
+    *   Approach 1 : Use PendingTransaction, which just like tools::wallet2::multisig_tx_set has a vector of pending txs and an unordered set of signers pub keys.
+    *   Approach 2 : Use the encrypted hex string generated with tools::wallet2::save_multisig_tx(). This approach is already used in WalletImpl::restoreMultisigTransaction.
+    * __________ Approach 1 __________
+    * param: multisig_ptxs -
+    * __________ Approach 2 __________
+    * param: multisig_sign_data -
+    * ________________________________
+    * return : true if enote is frozen, else false
+    * note: sets status error on fail
+    */
+    virtual bool isFrozen(std::size_t idx) const = 0;
+    virtual bool isFrozen(const std::string &key_image) const = 0;
+    virtual bool isFrozen(const PendingTransaction &multisig_ptxs) const = 0;
+    virtual bool isFrozen(const std::string multisig_sign_data) const = 0;
+    // QUESTION : Should we add the following function, which returns strings containing the key image for every frozen enote? I think it can be useful
+//    virtual std::vector<std::string> getAllFrozenEnotes() const = 0;
+    // TODO : Would this better fit in one of `Subaddress`/`SubaddressAccount` classes?
+    /**
+    * brief: createOneOffSubaddress - create a subaddress for given index
+    * param: account_index - major index
+    * param: address_index - minor index
+    */
+    virtual void createOneOffSubaddress(std::uint32_t account_index, std::uint32_t address_index) = 0;
+    /**
+    * brief: isDeprecated - check if wallet file format is deprecated
+    * return: true if wallet file is old format (before becoming JSON), else false
+    */
+    virtual bool isDeprecated() const = 0;
```

Comment:
What would you think about an attempt to get rid of this and similar very, very rarely used and "exotic" (but still needed) methods by introducing a call like `getWalletState` or similar that gives back a struct with a number of such info points?

Reply:
Sounds like a great idea to me.